version: '3'

services:

  postgres:
    container_name: postgres_container
    image: postgres
    ports: 
      - 5432:5432
    environment:
      POSTGRES_USER: username
      POSTGRES_PASSWORD: password
      POSTGRES_DB: postgres
      PGDATA: /data/postgres
    volumes:
      - $HOME/docker/volumes/postgres:/var/lib/postgresql/data
    networks:
      - postgres
    restart: unless-stopped


  pgadmin:
    container_name: pgadmin_container
    image: dpage/pgadmin4
    environment:
      PGADMIN_DEFAULT_EMAIL: pgadmin4@pgadmin.org
      PGADMIN_DEFAULT_PASSWORD: admin
      PGADMIN_CONFIG_SERVER_MODE: 'False'
    volumes:
       - pgadmin:/root/.pgadmin
    ports:
      - 5050:80
    networks:
      - postgres
    restart: unless-stopped


  app:
    container_name: netflix_container
    image: netflix:latest
    build:
      context: ./netflix_app
      dockerfile: Dockerfile
    volumes: 
      - /absolute/path/to/the/repo/netflix_app/data:/app/data
    networks:
      - postgres
    restart: on-failure


networks:
  postgres:  # name we are using inside this docker-compose
    external:
      name: postgres  # name outside the docker-compose


#  networks:
#    postgres:
#      driver: bridge


volumes:
    postgres:
    pgadmin:

